cfg/CFG.h

```diff
@@ -68,22 +68,39 @@ class BasicBlock final {
         counterInc("basicblocks");
     };
 
-    std::string toString(const core::GlobalState &gs) const;
-    std::string showRaw(core::Context ctx) const;
+    std::string toString(const core::GlobalState &gs, const CFG &cfg) const;
+    std::string showRaw(const core::GlobalState &gs, const CFG &cfg) const;
 };
 
 class CFGContext;
 
 class CFG final {
+public:
+    class UnfreezeCFGLocalVariables final {
+        CFG &cfg;
+
+    public:
+        UnfreezeCFGLocalVariables(CFG &cfg);
+
+        ~UnfreezeCFGLocalVariables();
+    };
+
     friend class CFGBuilder;
+    friend class LocalRef;
+    friend class UnfreezeCFGLocalVariables;
     /**
      * CFG owns all the BasicBlocks, and then they have raw unmanaged pointers to and between each other,
      * because they all have lifetime identical with each other and the CFG.
      */
-public:
     core::SymbolRef symbol;
     int maxBasicBlockId = 0;
     int maxRubyBlockId = 0;
+
+    /**
+     * Get the maximum local variable ID. Used to size vectors that contain an entry per LocalRef.
+     */
+    int maxVariableId() const;
```

Comment:
Is this only ever used to size vectors based on a CFG? If so, it might be worth renaming it to something like `numLocalVariables`, or `localVariablesSize`.